detox/test/e2e/15.urls.test.js

```diff
@@ -1,23 +1,46 @@
 describe('Open URLs', () => {
 
-  it('device.launchApp({{newInstance: true, url: url}) should launch app and trigger handling open url handling in app', async () => {
-    const url = 'detoxtesturlscheme://such-string';
-    await device.launchApp({newInstance: true, url: url});
-    await expect(element(by.text(url))).toBeVisible();
+  const withDefaultArgs = () => ({
+    url: 'detoxtesturlscheme://such-string',
+    launchArgs: undefined,
   });
 
-  it('device.openURL({url: url}) should trigger open url handling in app when app is in foreground', async () => {
-    const url = 'detoxtesturlscheme://such-string';
-    await device.launchApp({newInstance: true});
-    await device.openURL({url: url});
-    await expect(element(by.text(url))).toBeVisible();
+  const withAndroidSingleTaskActivityArgs = () => ({
+    url: 'detoxtesturlscheme.singletask://such-string',
+    launchArgs: Object.freeze({ androidSingleTaskActivity: true }),
```

Comment:
singleInstance

What is the reason for freezeing the object?

Reply:
why do you need `androidSingleTaskActivity` as a param here? isn't the custom url enough?

Reply:
I'm afraid the url is insufficient since we need to launch the app in single-instance mode if we really want to test this behaviour out.
As for `Object.freeze()`-ing, that is due to another bug I found; I've introduced a new unit test instead of `Object.freeze()`-ing here, check it out.